plugins/modules/network_compliance_workflow_manager.py

```diff
@@ -874,6 +876,107 @@ def get_want(self, config):
 
         return self
 
+    def get_compliance_report(self, run_compliance_params, mgmt_ip_instance_id_map):
+        """
+        Generate a compliance report for devices based on provided parameters.
+
+        This function fetches the compliance details for a list of devices specified
+        in the run_compliance_params. It maps the device UUIDs to their corresponding
+        management IPs, and then retrieves the compliance details for each device.
+
+        Args:
+            run_compliance_params (dict): Parameters for running compliance checks.
+                                          Expected to contain "deviceUuids" and optionally "categories".
+            mgmt_ip_instance_id_map (dict): Mapping of device management IPs to device UUIDs.
+
+        Returns:
+            dict: A dictionary with device management IPs as keys and lists of compliance details as values.
+        """
+
+        # Initialize the lists/dicts
+        final_response = {}
+        device_list = []
+        compliance_details_of_device_params = {}
+
+        # Iterate through each device UUID in the run compliance parameters
+        for device_uuid in run_compliance_params["deviceUuids"]:
+
+            # Find the corresponding device IP for the given device UUID
+            for ip, device_id in mgmt_ip_instance_id_map.items():
+                if device_uuid == device_id:
+                    device_ip = ip
+                    break
+
+            # Add the device IP to the device list
+            device_list.append(device_ip)
+
+            # Initialize the response list for the device IP if not already present
+            if device_ip not in final_response.keys():
+                final_response[device_ip] = []
+
+            # Check if categories are specified and fetch details for each category of the device
+            if "categories" in run_compliance_params.keys():
+                for category in run_compliance_params["categories"]:
+                    compliance_details_of_device_params["category"] = category
+                    compliance_details_of_device_params["device_uuid"] = device_uuid
```

Comment:
do we set device_uuid, and all inside for loop? Is it common for all the category?

Reply:
yes, so the new API that we are using is such that:
for a single device we can get all categories at once. But if we want category specific report, we have to do a API call for each category separately.